paper-api/src/main/java/io/papermc/paper/event/world/ExplodeEvent.java

```diff
@@ -0,0 +1,158 @@
+package io.papermc.paper.event.world;
+
+import org.bukkit.ExplosionResult;
+import org.bukkit.Location;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.ApiStatus;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+import java.util.List;
+
+/**
+ * Called when an explodes happen. this includes explosion who not affect blocks and are just visual.
+ */
+@ApiStatus.Experimental
+@NullMarked
+public class ExplodeEvent extends Event implements Cancellable {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    private final Location location;
+    private final List<Block> blocks;
+    private final ExplosionResult result;
+    private @Nullable Block block;
```

Comment:
No Block. It's always air, location is enough.

Reply:
Then remove the getBlock methods?
Just BlockState and Entity?

Reply:
yeah

Reply:
removed from the event class.